src/components/scenes/RequestScene.js

```diff
@@ -130,7 +134,8 @@ export class Request extends Component<Props, State> {
     const { edgeWallet, useLegacyAddress } = this.props
     let publicAddress = this.props.publicAddress
     let legacyAddress = this.props.legacyAddress
-    const abcEncodeUri = useLegacyAddress ? { publicAddress, legacyAddress } : { publicAddress }
+    let segwitAddress = this.props.segwitAddress
+    const abcEncodeUri = segwitAddress ? { segwitAddress, publicAddress } : useLegacyAddress ? { publicAddress, legacyAddress } : { publicAddress }
```

Comment:
Does `encodeUri` understand segwitAddress`? Can you verify that if you enter an amount, the QR code still uses the segwit address, and not the public address?

Reply:
No it does not.